opm/autodiff/StandardWells_impl.hpp

```diff
@@ -1248,4 +1248,304 @@ namespace Opm
+
+           // economic limits only apply for production wells.
+           if (wells_struct->type[w] != PRODUCER) {
+               continue;
+           }
+
+           // if no limit is effective here, then continue to the next well
+           if ( !econ_production_limits.onAnyEffectiveLimit() ) {
+               continue;
+           }
+           // for the moment, we only handle rate limits, not handling potential limits
+           // the potential limits should not be difficult to add
+           const WellEcon::QuantityLimitEnum& quantity_limit = econ_production_limits.quantityLimit();
+           if (quantity_limit == WellEcon::POTN) {
+               const std::string msg = std::string("POTN limit for well ") + well_name + std::string(" is not supported for the moment. \n")
+                                     + std::string("All the limits will be evaluated based on RATE. ");
+               OpmLog::warning(msg);
+           }
+
+           const WellMapType& well_map = well_state.wellMap();
+           typename WellMapType::const_iterator i_well = well_map.find(well_name);
+           assert(i_well != well_map.end()); // should always be found?
+           const int well_number = (i_well->second)[0];
+
+           if (econ_production_limits.onAnyRateLimit()) {
+               rate_limit_violated = checkRateEconLimits(econ_production_limits, well_state, well_number);
+           }
+
+           if (rate_limit_violated) {
+               if (econ_production_limits.endRun()) {
+                   const std::string warning_message = std::string("ending run after well closed due to economic limits is not supported yet \n")
+                                                     + std::string("the program will keep running after ") + well_name + std::string(" is closed");
+                   OpmLog::warning(warning_message);
+               }
+
+               if (econ_production_limits.validFollowonWell()) {
+                   OpmLog::warning("opening following on well after well closed is not supported yet");
+               }
+
+               if (well_ecl->getAutomaticShutIn()) {
+                   list_econ_limited.addShutWell(well_name);
+                   const std::string msg = std::string("well ") + well_name + std::string(" will be shut in due to economic limit");
+                   OpmLog::info(msg);
+               } else {
+                   list_econ_limited.addStoppedWell(well_name);
+                   const std::string msg = std::string("well ") + well_name + std::string(" will be stopped due to economic limit");
+                   OpmLog::info(msg);
+               }
+               // the well is closed, not need to check other limits
+               continue;
+           }
+
+           // checking for ratio related limits, mostly all kinds of ratio.
+           int worst_offending_connection = -1e4;
```

Comment:
This looks weird, in particular -1e4 is a double literal. I suggest that instead of passing worst_offending_connection and last_connection as output parameters for checkRatioEconLimits(), you let it return a tuple or a small struct containing the bool that is the current return value as well as those two parameters.